pkgs/development/libraries/gmock/default.nix

```diff
@@ -1,15 +1,13 @@
-{ stdenv, fetchurl, unzip, cmake}:
-
+{ stdenv, cmake, callPackage }:
+let
+  source = callPackage ./source.nix { };
```

Comment:
I took this pattern from pkgs/development/libraries/gtest/default.nix . Does that one use it better than I do?
Do you have an example of what you are proposing? I'm not sure I understand your proposal correctly.